apps/speed.c

```diff
@@ -1659,7 +1459,7 @@ int speed_main(int argc, char **argv)
         {"nistp256", NID_X9_62_prime256v1, 256},
         {"nistp384", NID_secp384r1, 384},
         {"nistp521", NID_secp521r1, 521},
-# ifndef OPENSSL_NO_EC2M
+#ifndef OPENSSL_NO_EC2M
```

Comment:
Should these CPP conditional lines be removed?

Reply:
I'd probably need to add some pre-checking code to avoid errors with openssl speed ecdsa in the no-ec2m builds. Not sure it is worth it.

Reply:
This would fail after the fetch.  I keep forgetting that about the EC algorithms.

Reply:
I had the exact same thought.  Considering that the presence of diverse algorithms is much more dynamic, having these guards in applications make less and less sense.  Generally speaking, this was already true since ENGINEs learned to do EVP level stuff, and `openssl speed` does take an `-engine` option.

However, I think this can be left for another PR, perhaps one that introduces the possibility to specify providers (and oh boy, I can see that lead to other mods, and we'll end up with a very different speed app)